### PR TITLE
IsItUp: Attempt to add support for domains containing unicode characters for #220

### LINKED ISSUE #220: IsItUp: Add support for domains containing unicode characters
Eg. ["is føtex.dk up"](https://duckduckgo.com/?q=is+f%C3%B8tex.dk+up)

- "expand føtex.dk"
- "whois føtex.dk".

Seems that their API might not support unicode. Needs further investigation!

https://duck.co/ia/view/is_it_up

### PR DESCRIPTION
**Pull request by rasikapohankar**
###### What does your Pull Request do (check all that apply)?
- [x] Improvement
    - [x] Enhancement: **`{IsItUp}: Include support for domains with Unicode characters`**

###### Description of changes
Modified regex to match Unicode characters, used Domain::PublicSuffix instead of regular expression to match and get root domain name, used Data::Validate::IP to ensure IP is valid IPv4 address, modified test file.

###### Which issues (if any) does this fix?

Fixes #220 

###### People to notify (@mention interested parties)
@MrChrisW @moollaza 

---

Instant Answer Page: https://duck.co/ia/view/is_it_up

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mrshu

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by edgesince84** [comment]
Thanks @rasikapohankar.  @mrshu is the maintainer, and he should be in here shortly to take a look!

File: lib/DDG/Spice/IsItUp.pm [hunk 1]
```diff
@@ -3,28 +3,35 @@ package DDG::Spice::IsItUp;
...
-        # return the domain and the root url if the domain is valid
-        if ($domain =~ $regex_domain){
-            return $root_url.$domain;
+        $ascii = domain_to_ascii($_[1].$_[2]);
+        $domain = $publicSuffix->get_root_domain($ascii);
+
+        if(!$domain) {  #if domain is undefined, it may be incorrect as a whole or the TLD was not found
+            $domain = $_[1].$_[2] if tld_exists(substr $_[2], 1);
         }
+        return if !$domain;
```
Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: `return unless $domain`

File: lib/DDG/Spice/IsItUp.pm [hunk 2]
```diff
@@ -3,28 +3,35 @@ package DDG::Spice::IsItUp;
...
-my $regex_ipv4 = qr/^(?:\d{1,3}\.){3}\d{1,3}$/;
-
 handle matches => sub {
+
+    my ($domain, $ascii);
+    my $publicSuffix = Domain::PublicSuffix->new();
+
     if ($_[2]) {
-        my $root_url = $_[1];
-        my $domain = $_[2];
```
Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: Any reason you removed these variables? IMO using named variables makes the code easier to read/follow.

File: lib/DDG/Spice/IsItUp.pm [hunk 3]
```diff
@@ -33,12 +40,12 @@ handle matches => sub {
...
+            $domain = $publicSuffix->get_root_domain($_[1]);
+            return if !$domain;
+            return domain_to_unicode($domain);
         }
     }
     return;
 };
 
-1;
+1;;
```
Comment (moollaza) [inline lib/DDG/Spice/IsItUp.pm]: Extra `;`?

**Comment by moollaza** [comment]
Deploying to Beta.

File: lib/DDG/Spice/IsItUp.pm (hunk 2 above)
Comment (rasikapohankar) [inline lib/DDG/Spice/IsItUp.pm]: @moollaza No, no reason. Agreed, I'll add them back :+1:

**Comment by mrshu** [comment]
Thanks a ton for your improvements @rasikapohankar.

It has been quite some time since I did something with Spice of any sort and so I do not think I will do a good job at maintaining this one further. In this PR you pretty much rewrote the whole Spice from scratch and so I think you should also maintain it in the future.

With regards to the PR, it looks good to me. I no longer know the protocol but if @moollaza is OK with merging this, so am I.

@moollaza if @rasikapohankar agrees with taking up the maintainership of this Spice, can you please take care of the transition? Also, please feel free to merge this PR if you are satisfied with it.

Thanks!

**Comment by rasikapohankar** [comment]
Thanks @mrshu, I'd love to be a maintainer for this IA! :)

**Comment by edgesince84** [comment]
@rasikapohankar I just made you the maintainer!  🎉

**Comment by rasikapohankar** [comment]
Thanks @edgesince84! Is it possible to have more than one maintainer for an IA?
Not related to this, I was just curious, what is the `Active` beside Dax's deployment update? Is it something GitHub added?

**Comment by moollaza** [comment]
> Is it possible to have more than one maintainer for an IA?

@rasikapohankar currently, no. 

> Not related to this, I was just curious, what is the Active beside Dax's deployment update? Is it something GitHub added?

I believe so, to tell you the current status of the deployment.

@rasikapohankar Thanks! This LGTM :+1:

**Pull request merged**